### PR TITLE
modules: trusted-firmware-m: fix dependencies for isolation level

### PR DESCRIPTION
**Pull request by ioannisg**
Do not allow the TFM_ISOLATION_LEVEL to be non-hidden option,
if the IPC (PSA_API) option is not set, since, in that case,
only Isolation Level 1 is supported.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
This will probably fail, now, since stm32 l552ze_2 forces TFM isolation level to 2, while it still allows the psa_level1 sample to build and run.

File: modules/trusted-firmware-m/Kconfig [hunk 1]
```diff
@@ -120,17 +120,19 @@ config TFM_CMAKE_BUILD_TYPE_DEBUG
...
+	default 1 if TFM_PROFILE_TYPE_SMALL || !TFM_IPC
 	default 2 if TFM_PROFILE_TYPE_MEDIUM
 	default 3 if TFM_PROFILE_TYPE_LARGE
 	help
 	  Manually set the required TFM isolation level. Possible values are
 	  1,2 or 3; the default is set by build configuration. When TF-M
 	  Profile option is supplied, do not allow manual setting of the
 	  isolation level, as it is determined by the profile setting.
+	  The isolation level is to be fixed to 1 if TF-M is not build with
+	  PSA_API (IPC) support.
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:135]:
```suggestion
	  Isolation levels 2 and 3 require PSA_API (IPC) support.
```

Comment (erwango) [reply modules/trusted-firmware-m/Kconfig:135]:
```suggestion
	  As isolation levels 2 and 3 require PSA_API (IPC) support, force level 1 when not enabled.
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by ioannisg** [comment]
@erwango @oyvindronningstad thanks for the review - text updated.

**Review by oyvindronningstad — APPROVED** [review]

**Comment by ioannisg** [comment]
I added a commit that removes the forcing of TF-M isolation level to 2 for STM32 boards that support TFM - there does not seem to be any need for this, and the proof is the fact that psa_level1 sample could run successfully for the STM32L562 board (there was a Kconfig overlay that was explicitly setting the isolation level to 1)

**Review by erwango — APPROVED** [review]

**Comment by ioannisg** [comment]
Will test psa_level1 on nucleo_l552ze_q before merging it.

**Comment by ioannisg** [comment]
Need an overlay for the nucleo board. Wil add a re-run CI.

**Comment by ioannisg** [comment]
Tested the integration samples  on nucleo_l552ze_q_ns

**Pull request merged**